server/src/handlers/users.rs

```diff
@@ -7,61 +7,262 @@ use axum::{
 use futures::TryStreamExt;
 use mongodb::{
     bson::oid::ObjectId,
-    results::{DeleteResult, InsertOneResult, UpdateResult},
+    results::{DeleteResult, UpdateResult},
 };
+use serde::{Deserialize, Serialize};
 
+use crate::auth::middleware::{AdminAuth, AuthorizationError, SelfOrAdminAuth};
+use crate::models::Token;
 use crate::{
+    auth,
     db_connection::DBClient,
     error::CheeseError,
     models::{User, UserList},
 };
 
 pub async fn fetch_users(
     Extension(db_client): Extension<DBClient>,
+    _auth: AdminAuth,
 ) -> Result<Json<UserList>, CheeseError> {
     let cursor = db_client.user_repo.get_all_users().await?;
-    Ok(Json(UserList {
-        users: cursor.try_collect().await?,
-    }))
+
+    let mut users: Vec<User> = cursor.try_collect().await?;
+    for u in &mut users {
+        u.hashed_password = String::new();
+    }
+
+    Ok(Json(UserList { users }))
 }
 
 pub async fn fetch_user(
     Path(id): Path<String>,
     Extension(db_client): Extension<DBClient>,
+    auth: SelfOrAdminAuth,
 ) -> Result<Json<Option<User>>, CheeseError> {
-    let res = db_client
-        .user_repo
-        .read_user(ObjectId::from_str(&id)?)
-        .await?;
-    Ok(Json(res))
-}
+    let id = ObjectId::from_str(&id)?;
+
+    auth.user_by_id(&id)?;
+
+    let mut res = db_client.user_repo.get_user_by_id(id).await?;
+
+    if let Some(ref mut user) = res {
+        user.hashed_password = String::new();
```

Comment:
we could then also reuse that method here

Reply:
idem